checkov/secrets/plugins/entropy_keyword_combinator.py

```diff
@@ -34,26 +109,125 @@ def analyze_line(
             **kwargs: Any,
     ) -> set[PotentialSecret]:
-        """
-        This method first runs the keyword plugin. If it finds a match - it runs the entropy scanners, and if
-        one of the entropy scanners find a match (on a line which was already matched by keyword plugin) - it is returned.
-        for source code files run and merge the two plugins.
-        """
         is_iac = f".{filename.split('.')[-1]}" not in SOURCE_CODE_EXTENSION
+        filetype = determine_file_type(filename)
+        value_keyword_regex_to_group = REGEX_VALUE_KEYWORD_BY_FILETYPE.get(filetype, FileType.YAML)
+        secret_keyword_regex_to_group = REGEX_VALUE_SECRET_BY_FILETYPE.get(filetype, FileType.YAML)
+
         if len(line) <= MAX_LINE_LENGTH:
             if is_iac:
-                keyword_matches = self.keyword_scanner.analyze_line(filename, line, line_number, **kwargs)
-                if keyword_matches:
-                    for entropy_scanner in self.high_entropy_scanners_iac:
-                        matches = entropy_scanner.analyze_line(filename, line, line_number, **kwargs)
-                        if matches:
-                            return matches
+                return self.analyze_iac_line(
+                    filename=filename,
+                    line=line,
+                    line_number=line_number,
+                    context=context,
+                    raw_context=raw_context,
+                    value_pattern=value_keyword_regex_to_group,
+                    secret_pattern=secret_keyword_regex_to_group,
+                    kwargs=kwargs
+                )
             else:
-                for entropy_scanner in self.high_entropy_scanners:
-                    matches = entropy_scanner.analyze_line(filename, line, line_number, **kwargs)
-                    if matches:
-                        return matches
+                return self.detect_secret(self.high_entropy_scanners, filename, line, line_number, **kwargs)
+        return set()
+
+    def analyze_iac_line(
+            self,
+            filename: str,
+            line: str,
+            line_number: int = 0,
+            context: CodeSnippet | None = None,
+            raw_context: CodeSnippet | None = None,
+            value_pattern: Optional[dict[Pattern, int]] = None,
+            secret_pattern: Optional[dict[Pattern, int]] = None,
+            **kwargs: Any,
+    ) -> set[PotentialSecret]:
+        secrets = set()
+
+        # classic key-value pair
+        keyword_on_key = self.keyword_scanner.analyze_line(filename, line, line_number, **kwargs)
+        if keyword_on_key:
+            return self.detect_secret(self.high_entropy_scanners_iac, filename, line, line_number, **kwargs)
+
+        # not so classic key-value pair, from multiline, that is only in an array format.
+        # The scan is one-way backwards, so no duplicates expected.
+
+        elif context is not None and raw_context is not None:
```

Comment:
@marynaKK Everything below this line is for yaml ONLY. Need to make sure that non yaml files are not affected by this change at all - please make sure that non-yaml files are run the same way they did before. It is not clear from the flow that this is what would happen.

Reply:
Checked if it’s yml but forgot to use it - great to have a review!!!

Reply:
added some logic - so now only yaml should be parsed.